RubeesEat/Model/Person.cs

```diff
@@ -23,8 +23,38 @@ public Person(Guid id, string firstName, string lastName)
         LastName = lastName;
     }
 
-    private bool IsValidName(params string[] words)
+    private static bool IsValidName(params string[] words)
     {
         return words.All(s => Regex.IsMatch(s, @"^[a-zA-Z]+$"));
     }
+    
+    public override bool Equals(object? obj)
+    {
+        if (obj is not Person other)
+            return false;
+
+        return Id == other.Id;
```

Comment:
Now we have two different equality implementations for domain objects. One where we compare ID + properties (see `Bill`) and this one where we compare just the ID. We should decide on one implementation for domain objects and use it everywhere.

Using `IEqualityComparer<T>` we can implemented different equality comparison methods outside of the domain object class. I would suggest removing the equality from the domain objects entirely. This will revert them to reference equality. We can then implement `IEqualityComparer<T>` to have different equality contracts. So we would have one for the ID equality and one for full equality. Try to find a way that we don't have to write the ID equality comparer for all domain object separately but that we can reuse the same class for all domain objects.